js/gateio.js

```diff
@@ -1089,19 +1089,65 @@ module.exports = class gateio extends Exchange {
         return underlyings;
     }
 
-    prepareRequest (market) {
+    prepareRequest (market = undefined, type = undefined, setMarginType = false, stop = false, multiOrder = false, params = {}) {
```

Comment:
Tbh, I don't think that it was a good idea to introduce a large bulk method that prepares all requests, since their formats and requirements differ a lot. We're doing a lot of checks for margin, stop orders and other things just to fetch the tickers. I'd rather not introduce `prepareRequest` at all and would write out these checks internally in each related method with just the specifics of the endpoint in question.

Reply:
> We're doing a lot of checks for margin, stop orders and other things just to fetch the tickers. 

Actually, if `setMarginType` is set to false, that's the only extra check you do, if it's false the code moves on, if you meant literally.

> makes unifying different exchanges harder

This is only a method for gateio, because their api has about 4 or 5 different patterns for requests, to make it easier not to forget parameters and have defaults and cross/isolated set correctly when setting up a new method. It's a safety feature. I have split it into 2 methods though.

> I'd rather not introduce prepareRequest

prepareRequest was already there, but it covered less

> I'd rather suggest removing prepareRequest entirely.

The purpose of it is to make it easier not to forget parameters and have defaults and cross/isolated set correctly consistently when setting up a new method. It's a safety feature.